src/kubernetes/api_watcher.rs

```diff
@@ -482,6 +482,74 @@ mod tests {
                     }),
                 ],
             ),
+            // Non-desync Stream Error
+            (
+                Box::new(|when, then| {
+                    when.method(GET).path("/api/v1/pods");
+                    then.status(500);
+                }),
+                vec![Box::new(|item| {
+                    let error = item.unwrap_err();
+                    match error {
+                        watcher::stream::Error::Other { .. } => {}
```

Comment:
Would be great if we actually destructure and assert this error more knowingly - i.e. assert that it's an http error under the hood. One of the goals here is to ensure the errors are propagated correctly, and we get exactly the errors we expect to get, rather than "some" error that's not a desync.